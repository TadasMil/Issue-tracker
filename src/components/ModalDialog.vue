<script setup lang="ts" name="ModalDialog">
import { injectModal } from '@/composables/useModal'
import { computed } from 'vue'

import BaseModal from './BaseModal.vue'

const modal = injectModal()

const title = computed(() => modal.value?.title)
const visible = computed(() => modal.value?.visible)
</script>

<template>
  <teleport to="body">
    <BaseModal
      v-if="modal?.visible"
      v-slot="slotProps"
      :title="title!"
      :visible="visible!"
      :close="modal.close!"
    >
      <slot :close="slotProps.close" />
    </BaseModal>
  </teleport>
</template>
