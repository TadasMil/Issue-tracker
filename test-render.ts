import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
