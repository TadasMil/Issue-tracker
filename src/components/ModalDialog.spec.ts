import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest'
import { render } from '../../test-render'

import ModalDialog from './ModalDialog.vue'

const closeFn = vi.fn()

vi.mock('@/composables/useModal', () => ({
  injectModal: () => ({
    title: 'Modal Title',
    visible: true,
    close: closeFn,
  }),
}))

describe('ModalDialog', () => {
  beforeEach(() => {
    const el = document.createElement('div')
    el.id = 'modal'
    document.body.appendChild(el)
  })

  afterEach(() => {
    document.body.outerHTML = ''
  })

  it('should render configured modal component', () => {
    const { html } = render(ModalDialog, {
      global: {
        stubs: ['BaseModal'],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
