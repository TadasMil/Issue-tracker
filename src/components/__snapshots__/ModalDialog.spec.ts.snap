// Vitest Snapshot v1

exports[`ModalDialog > should render configured modal component 1`] = `
"<!--teleport start-->
<!--teleport end-->"
`;
